Copy module (#1902)
Codecademy/docs: content/pandas/concepts/dataframe/terms/copy/copy.md

---
Title: '.copy()'
Description: 'Returns a copy of a DataFrame or Series.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Data Structures'
  - 'Pandas'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

In Pandas, **`.copy()`** is a method that creates a copy of a DataFrame or a Series. This method returns a fully independent copy of DataFrame or a shallow copy (see Shallow vs. Deep Copy section).

## Syntax

```pseudo
df_copy = df.copy()
```
The boolean parameter `deep` specifies whether `.copy()` should create a deep or a shallow copy. The default value is `deep=True`, which enforces a deep copy.                                                            

## Shallow vs. Deep Copy

Note that the designation of the `deep` parameter can have impacts on the resulting copy as well as the original DataFrame.

- In a shallow copy, the new object points to the same data as the original object, and any changes made to the copy will affect the original object. Changes made to the original dataframe will also be reflected in the shallow copy. By default, the `.copy()` method creates a deep copy. 

> - In a deep copy, a new object is created with a completely new set of data that is identical to the original data. Changes made to the copied object will not affect the original object and vice versa, even if the original data is modified. 

Reviewer: ```suggestion
- In a deep copy, a new object is created with a completely new set of data that is identical to the original data. Changes made to the copied object will not affect the original object and vice versa. 
```

---
Title: '.copy()'
Description: 'Returns a DataFrame object with rows or columns removed based on column or index names.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Data Structures'
  - 'Pandas'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

In Pandas, .copy() is a method that creates a copy of a DataFrame or a Series. This method either returns a fully independent copy of dataframe or a shallow copy (see Shallow vs. Deep Copy section).

## Syntax
```pseudo
df_copy = df.copy()
```

> | Parameter Name | Data Type                | Usage                                                                                                                             |

Reviewer 2: Yes, please remove them.

| :------------: | ------------------------ | --------------------------------------------------------------------------------------------------------------------------------- |
|     `deep`     | bool                     | This argument specifies whether the copy should be a deep copy or a shallow copy. If deep=True, then the copy is a deep copy      |
|    `inplace`   | bool                     | If inplace=False (the default), the copy is made to a new object, and the original object is left unchanged.                      |
|     `meta`     | python dict              | This argument is used to specify metadata to be attached to the copy                                                              |

## Shallow vs. Deep Copy

> Notice that .copy() method has a deep parameter that can dictate the behavior of the resulting dataframe. We can either make a shallow or a deep copy of pandas dataframe.

Reviewer 2: ```suggestion
Note that the designation of the `deep` parameter can have impacts on the resulting copy as well as the original DataFrame.

```

> 1. In a shallow copy, the new object points to the same data as the original object, and any changes made to the copy will affect the original object. Changes made to the original dataframe will also be reflected in the shallow copy. By default, .copy() creates a shallow copy. But you can set deep=False to enforce a shallow copy.

Reviewer 2: ```suggestion
- In a shallow copy, the new object points to the same data as the original object, and any changes made to the copy will affect the original object. Changes made to the original dataframe will also be reflected in the shallow copy. By default, the `.copy()` method creates a deep copy. 
```

> 

Reviewer 2: ```suggestion
```

> 2. In a deep copy, a new object is created with a completely new set of data that is identical to the original data. Changes made to the copied object will not affect the original object and vice versa, even if the original data is modified. You can set deep=True to enforce a deep copy.

Reviewer 2: ```suggestion
- In a deep copy, a new object is created with a completely new set of data that is identical to the original data. Changes made to the copied object will not affect the original object and vice versa, even if the original data is modified. 
```

## Example:

> ```py

Reviewer 2: I would recommend that you remove the output comments. Follow the print statement on line 57 with the other two print statements and close the code block. Then have a line of text "The output of the code above will be:" and follow with a ```shell block that just contains the three dataframes.

import pandas as pd

> # create a DataFrame

Reviewer 2: ```suggestion
# Create a DataFrame
```

Reviewer 2: always start comments with a capital
